AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainActivity.kt

```diff
@@ -160,32 +160,33 @@ class MainActivity : SalesforceActivity() {
     private fun handleAppsListItemClick(position: Int) {
         Log.d(TAG, "Apps list item clicked, position: " + position)
         val appName = appsListView?.adapter?.getItem(position) as String
-        var appPackageName = ""
-        var appComponentName = ""
-        when (appName) {
-            MOBILE_SYNC_EXPLORER -> {
-                appPackageName = MOBILE_SYNC_EXPLORER_PACKAGE
-                appComponentName = MOBILE_SYNC_COMPONENT_NAME
-            }
-            REST_EXPLORER -> {
-                appPackageName = REST_EXPLORER_PACKAGE
-                appComponentName = REST_EXPLORER_COMPONENT_NAME
-            }
-            ACCOUNT_EDITOR -> {
-                appPackageName = ACCOUNT_EDITOR_PACKAGE
-                appComponentName = ACCOUNT_EDITOR_COMPONENT_NAME
+        val spAppPackageName = when (appName) {
+            MOBILE_SYNC_EXPLORER -> MOBILE_SYNC_EXPLORER_PACKAGE
+            REST_EXPLORER -> REST_EXPLORER_PACKAGE
+            ACCOUNT_EDITOR -> ACCOUNT_EDITOR_PACKAGE
+            else -> null
+        }
+        Log.d(TAG, "Launching SP app ${appName} with package ${spAppPackageName}")
+        if (spAppPackageName != null) {
+            SalesforceSDKManager.getInstance().idpManager?.let { idpManager ->
+                idpManager.kickOffIDPInitiatedLoginFlow(this, spAppPackageName,
```

Comment:
Basic way an app could show the status updates to the user (toasts). But other approaches are possible (progress bar etc).